codalab/model/mysql_model.py

```diff
@@ -2,13 +2,44 @@
 MySQLModel is a subclass of BundleModel that stores metadata on a MySQL
 server that it connects to with the given connect parameters.
 '''
-from sqlalchemy import create_engine
+from sqlalchemy import (
+    create_engine,
+    event,
+    exc,
+)
+from sqlalchemy.pool import Pool
 
 from codalab.model.bundle_model import BundleModel
 from codalab.common import (
     UsageError,
 )
 
+@event.listens_for(Pool, "checkout")
+def ping_connection(dbapi_connection, connection_record, connection_proxy):
+    '''
+    Checkout listener that ping the connection to check if it is alive.
+    If there is a problem (i.e. "MySQL server has gone away"), will raise a
+    sqlalchemy.DisconnectionError, which is caught internally by SQLALchemy and
+    forces up to three retries from the connection pool.
```

Comment:
The three retries is built into sqlalchemy, `DisconnectionError` is caught internally up to 3 times. The link to the documentation covers this I think